multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/service/ObjectStoreServiceInfoCreator.java

```diff
@@ -6,57 +6,104 @@
 import java.util.Base64;
 import java.util.Map;
 
+import org.cloudfoundry.multiapps.controller.web.Constants;
 import org.cloudfoundry.multiapps.controller.web.Messages;
 import org.jclouds.domain.Credentials;
 import org.jclouds.googlecloud.GoogleCredentialsFromJson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.google.common.base.Supplier;
 
 import io.pivotal.cfenv.core.CfService;
 
 public class ObjectStoreServiceInfoCreator {
 
-    private static final String OBJECT_STORE_AWS_PLAN = "s3-standard";
-    private static final String OBJECT_STORE_AZURE_PLAN = "azure-standard";
-    private static final String OBJECT_STORE_ALICLOUD_PLAN = "oss-standard";
-    private static final String OBJECT_STORE_GCP_PLAN = "gcs-standard";
+    private static final Logger LOGGER = LoggerFactory.getLogger(ObjectStoreServiceInfoCreator.class);
 
     public ObjectStoreServiceInfo createServiceInfo(CfService service) {
-        String plan = service.getPlan();
         Map<String, Object> credentials = service.getCredentials()
                                                  .getMap();
-        switch (plan) {
-            case OBJECT_STORE_AWS_PLAN:
-                return createServiceInfoForAws(credentials);
-            case OBJECT_STORE_AZURE_PLAN:
-                return createServiceInfoForAzure(credentials);
-            case OBJECT_STORE_ALICLOUD_PLAN:
-                return createServiceInfoForAliCloud(credentials);
-            case OBJECT_STORE_GCP_PLAN:
-                return createServiceInfoForGcpCloud(credentials);
-            default:
-                throw new IllegalStateException(Messages.UNSUPPORTED_SERVICE_PLAN_FOR_OBJECT_STORE);
+        if (isAliCloudIaas(credentials)) {
+            LOGGER.info(Messages.CREATING_OBJECT_STORE_FOR_ALI_CLOUD);
+            return createServiceInfoForAliCloud(credentials);
         }
+        if (isAwsIaas(credentials)) {
+            LOGGER.info(Messages.CREATING_OBJECT_STORE_FOR_AWS);
+            return createServiceInfoForAws(credentials);
+        }
+        if (isAzureIaas(credentials)) {
+            LOGGER.info(Messages.CREATING_OBJECT_STORE_FOR_AZURE);
+            return createServiceInfoForAzure(credentials);
+        }
+        if (isGcpIaas(credentials)) {
+            LOGGER.info(Messages.CREATING_OBJECT_STORE_FOR_GCP);
+            return createServiceInfoForGcpCloud(credentials);
+        }
+        throw new IllegalStateException(Messages.ERROR_BUILDING_OBJECT_STORE_CONFIGURATION);
```

Comment:
Do we depend now on order of checks in order to create initialize proper objectstore client? 
Other issue is that some of these parameters could be changed in bindings to null or missing at all in future and we will fail even if some of these properties are not required for our client at all, so the code will be more fragile with this logic right now.

Reply:
as discussed, refactored